### PR TITLE
Fetch token provided by app

### PR DESCRIPTION
**Pull request by sampettersson**

### CODE REVIEW

File: src/AppTokenRetrival.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+export const AppTokenRetrival = ({ children }) => (
+  <StorageContainer>
+    {(storageState) => (
+      <Mount
+        on={() => {
+          if (location.hash.includes('token=')) {
+            const token = location.hash.replace('#token=', '')
+            console.log('setting token to', token)
+            apolloClient!.subscriptionClient.close(true, true)
+            storageState.setToken(token)
```
Comment (palmenhq) [inline src/AppTokenRetrival.tsx]: shouldn't this happen in the reverse order?

File: src/AppTokenRetrival.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import { StorageContainer } from './utils/StorageContainer'
+
+export const AppTokenRetrival = ({ children }) => (
+  <StorageContainer>
+    {(storageState) => (
+      <Mount
+        on={() => {
+          if (location.hash.includes('token=')) {
+            const token = location.hash.replace('#token=', '')
+            console.log('setting token to', token)
```
Comment (palmenhq) [inline src/AppTokenRetrival.tsx]: 🙈

File: src/AppTokenRetrival.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import { apolloClient } from './client/apolloClient'
+import { StorageContainer } from './utils/StorageContainer'
+
+export const AppTokenRetrival = ({ children }) => (
+  <StorageContainer>
+    {(storageState) => (
+      <Mount
+        on={() => {
+          if (location.hash.includes('token=')) {
+            const token = location.hash.replace('#token=', '')
```
Comment (palmenhq) [inline src/AppTokenRetrival.tsx]: Could we do this with react router instead? Would make me feel better

File: src/AppTokenRetrival.tsx (hunk 1 above)
Comment (sampettersson) [reply src/AppTokenRetrival.tsx]: hmm, we do in this order in other places 🤔

Comment (sampettersson) [reply src/AppTokenRetrival.tsx]: but it makes more sense to set the token first, and that worked, so doing that instead 👨‍🔬

File: src/AppTokenRetrival.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+            if (location.hash.includes('token')) {
+              var params = {}
+
+              location.hash
+                .substring(1)
+                .split('&')
+                .map((hk) => {
+                  let param = hk.split('=')
+                  params[param[0]] = param[1]
+                })
```
Comment (palmenhq) [inline src/AppTokenRetrival.tsx]: This is a bit nit but using `Array#reduce` would be prettier ;) but i guess this is _acceptable_ although i think `forEach` would be more appropriate if we wouldn't want to use reduce for some reason

File: src/AppTokenRetrival.tsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+export const AppTokenRetrival = ({ children }) => {
+  const location = useLocation()
+
+  return (
+    <StorageContainer>
+      {(storageState) => (
+        <Mount
+          on={() => {
+            if (location.hash.includes('token')) {
+              var params = {}
```
Comment (palmenhq) [inline src/AppTokenRetrival.tsx]: `let`*

Comment (palmenhq) [reply src/AppTokenRetrival.tsx]: But most preferrably `Array#reduce` + `const`

File: src/AppTokenRetrival.tsx (hunk 4 above)
Comment (sampettersson) [reply src/AppTokenRetrival.tsx]: hahah was to quick copying from stackoverflow, this is now a reduce hehe

File: src/AppTokenRetrival.tsx (hunk 5 above)
Comment (sampettersson) [reply src/AppTokenRetrival.tsx]: haha muskelminne, in Swift a let is immutable but and a var is mutable 😭

**Pull request merged**